### PR TITLE
Type safety: implement enums

### PR DESCRIPTION
**Pull request by Ovifer13**
# Description

Relates to issue: **ZEUS-1262**

Implement an enum file where all enum variables are stored and can be used in all components.

This pull request is categorized as a:

- [ ] New feature
- [ ] Bug fix
- [x] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance 
- [ ] Other

## Checklist
- [x] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [x] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [x] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [x] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] LND (REST)
- [ ] LND (Lightning Node Connect)
- [ ] Core Lightning (c-lightning-REST)
- [ ] Core Lightning (Spark)
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: components/Channels/Tag.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { View } from 'react-native';
 import { Body } from '../../components/text/Body';
 import { Row } from '../../components/layout/Row';
 import { Spacer } from '../../components/layout/Spacer';
-import { Status } from '../../views/Channels/ChannelsPane';
+import { Status } from '../../enums';
```
Comment (kaloudis) [inline components/Channels/Tag.tsx:6]: this shouldn't have changed

File: enums/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+export enum FeeMethod {
+    fixed = 'fixed',
+    percent = 'percent'
+}
+
+export enum AddressType {
+    _0 = '0',
+    _1 = '1',
+    _4 = '4'
```
Comment (kaloudis) [inline enums/index.ts:9]: we should probably elaborate on the types here

**Comment by kaloudis** [comment]
Needs a rebase, some small changes, and of course some thorough testing, but really loving the idea and initiative here. Quite promising.

File: components/Channels/ChannelItem.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { View } from 'react-native';
 import { Body } from '../../components/text/Body';
 import { BalanceBar } from '../../components/Channels/BalanceBar';
 import { Row } from '../../components/layout/Row';
-import { Status } from '../../views/Channels/ChannelsPane';
+import { Status } from '../../enums';
```
[the remark this answers is not in the document]
Comment (Ovifer13) [reply components/Channels/ChannelItem.tsx:7]: Why not? Status was an enum inside `ChannelsPane` that's why I did the change.

File: enums/index.ts (hunk 2 above)
Comment (Ovifer13) [reply enums/index.ts:9]: I considered `zero`, `one`, and `four`. what do you think?

Comment (kaloudis) [reply enums/index.ts:9]: p2wk: 0,
np2wkh: 1,
p2tr: 4

OR spelled out

WITNESS_PUBKEY_HASH: 0,
NESTED_PUBKEY_HASH: 1,
TAPROOT_PUBKEY: 4

**Pull request closed without merging**